flytepropeller/pkg/controller/nodes/array/handler_test.go

```diff
@@ -624,7 +624,7 @@ func TestHandleArrayNodePhaseExecuting(t *testing.T) {
 			eventRecorder := newBufferedEventRecorder()
 
 			nodeSpec := arrayNodeSpec
-			nodeSpec.ArrayNode.Parallelism = uint32(test.parallelism)
+			nodeSpec.ArrayNode.Parallelism = int64(test.parallelism)
```

Comment:
can we add some unit tests here to validate that the ExecutionContext parallelism is correctly applied and incremented?